test/testdata/infer/control_flow/instance_of.rb

```diff
@@ -0,0 +1,17 @@
+# typed: true
+
+class A; end;
+class B; end;
+class C; end;
+
+extend T::Sig
+
+sig {params(x: T.any(A, B, C)).void}
+def baz(x)
```

Comment:
please add following test
```
# typed: true
x = "s"
if x.instance_of?(Object)
  puts x
else 
  puts x
end
```

Reply:
Oh, this is interesting. We discussed offline and realized that the solution is harder to implement than just doing this.

Will comment on the original ticket